src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java

```diff
@@ -0,0 +1,154 @@
+        Dog editedDog = new DogBuilder((Dog) firstIdEntity.getValue()).build();
+        EditDogCommand.EditDogDescriptor descriptor = new EditDogDescriptorBuilder(editedDog).build();
+        EditDogCommand editDogCommand = new EditDogCommand(firstIdEntity.getKey(), descriptor);
+
+        String expectedMessage = String.format(EditDogCommand.MESSAGE_EDIT_DOG_SUCCESS, editedDog);
+
+        Model expectedModel = new ModelManager(new Database(model.getDatabase()), new UserPrefs());
+        expectedModel.setEntity(firstIdEntity.getKey(), editedDog);
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(firstIdEntity.getKey()));
+
+        assertCommandSuccess(editDogCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_ownerIdSpecified_success() {
+        Dog toEditDog = (Dog) model.getEntity(ID_TWO);
+
+        DogBuilder dogInList = new DogBuilder(toEditDog);
+        Dog editedDog = dogInList.withName(VALID_NAME_BELL).withBreed(VALID_BREED_BELL)
+                .withTags(VALID_TAG_FRIENDLY).build();
+
+        EditDogCommand.EditDogDescriptor descriptor = new EditDogDescriptorBuilder().withName(VALID_NAME_BELL)
+                .withBreed(VALID_BREED_BELL).withTags(VALID_TAG_FRIENDLY).build();
+        EditDogCommand editEntityCommand = new EditDogCommand(ID_TWO, descriptor);
+
+        String expectedMessage = String.format(EditDogCommand.MESSAGE_EDIT_DOG_SUCCESS, editedDog);
+
+        Model expectedModel = new ModelManager(new Database(model.getDatabase()), new UserPrefs());
+        expectedModel.setEntity(ID_TWO, editedDog);
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(ID_TWO));
+
+        assertCommandSuccess(editEntityCommand, model, expectedMessage, expectedModel);
+    }
+
+    /*@Test
+    public void execute_someFieldsSpecified_success() {
+        Dog toEditDog = (Dog) model.getEntity(ID_TWO);
+
+        DogBuilder dogInList = new DogBuilder(toEditDog);
+        Dog editedDog = dogInList.withOwnerID(ID_THREE).build();
+
+        EditDogCommand.EditDogDescriptor descriptor = new EditDogDescriptorBuilder()
+                .withOwnerId(ID_THREE).build();
+        EditDogCommand editEntityCommand = new EditDogCommand(ID_TWO, descriptor);
+
+        String expectedMessage = String.format(EditDogCommand.MESSAGE_EDIT_DOG_SUCCESS, editedDog);
+
+        Model expectedModel = new ModelManager(new Database(model.getDatabase()), new UserPrefs());
+        expectedModel.setEntity(ID_TWO, editedDog);
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(ID_TWO));
+
+        Owner toEditOwner = (Owner) model.getEntity(ID_ONE);
+        Set<Integer> newDogIdSet = new HashSet<>(toEditOwner.getDogIdSet());
+        newDogIdSet.remove(ID_TWO);
+        model.setEntity(ID_ONE, new Owner(toEditOwner.getName(), toEditOwner.getPhone(), toEditOwner.getEmail(),
+                toEditOwner.getAddress(), toEditOwner.getTags(), newDogIdSet));
```

Comment:
Also, in the copy-paste process, you might have not spent the time to understand what each of the test are doing, there is an `expectedModel` which we modify by hand in the most crude way to keep things simple and avoid errors, while the `model` is the target that we let the `EditDogCommand` execute on and therefore modify.